lib/ast/primitive.ts

```diff
@@ -971,16 +971,21 @@ Stream.Timer.prototype.isTimer = true;
 export class AtTimerStream extends Stream {
     time : Value[];
     expiration_date : Value|null;
+    once : Value|false;
```

Comment:
Wait this type doesn't make sense. Ast.Value is not a boolean, it's an object, so the default value should be `new Ast.BooleanValue(false)`

Reply:
Would it be fine to just use the `boolean` type then? Or is it preferred to use the object

Reply:
It's preferred to use the object. If you use boolean, you need to enforce that the Ast.Value is a constant somewhere, or you'll crash when you call `once.value.toJS()` in `FunctionCallExpression.toLegacy`. Ast.Value is the base class of all scalar expressions in ThingTalk, which can be constants or can be computed expressions. In the syntax, a function can be called with arbitrary values (provided the types match) so you can't statically convert to a boolean. Similarly in the compiler you need to call compileValue so the compiler generates the appropriate JS code for the value.